main.cpp

```diff
@@ -1,14 +1,34 @@
+#include <vector>
 #include <iostream>
-#include "functions.cpp"
+#include "smart_home.h"
+
 int main()
 {
-    srand(time(0));
-    std::vector<Product> products;
-    for (int i = 0; i < 10; i++)
+    SmartHome smartHome;
+    try
     {
-        products.push_back(Product());
+        smartHome.GetDataFromXml("../data.xml");
+        
+        //Device device("BathroomDoor", "Door", true);
+        //Sensor presence("PresenceSensor", "Presence", 0);
+        
+        // Room *living = smartHome.GetRoom("LivingRoom");
+        // living->AddSensor(presence);
+        // living->AddDevice(device);
+
+        Sensor *lightSensor = smartHome.GetRoom("LivingRoom")->GetSensor("LightSensor");
+        float val = lightSensor->GetValue();
```

Comment:
you should check if your lightSensor is not null pointer

Reply:
Correct